app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -446,13 +474,30 @@ private void raiseMessage(NotificationMessage message, boolean showTop) {
         }
         
         //either way
+        setLoginBoxesColor(R.color.cc_attention_negative_color, true);
+        username.setCompoundDrawablesWithIntrinsicBounds(getResources().getDrawable(R.drawable.icon_user_attnneg),  null, null, null);
+        password.setCompoundDrawablesWithIntrinsicBounds(getResources().getDrawable(R.drawable.icon_lock_attnneg),  null, null, null);
+        loginButton.setBackgroundColor(getResources().getColor(R.color.cc_attention_negative_bg));
+        loginButton.setTextColor(getResources().getColor(R.color.cc_attention_negative_text));
+
         errorBox.setVisibility(View.VISIBLE);
         errorBox.setText(toastText);
         
         Toast.makeText(this,toastText, Toast.LENGTH_LONG).show();
     }
-    
-    
+
+    /**
+     * Sets the login boxes (user/pass) to the given color.
+     * @param colorIdOrCode Color ID of resource or color code
+     * @param isColorId Calls getResources.getColor if set, uses the given colorIdOrCode as the color code if not
+     */
+    private void setLoginBoxesColor(int colorIdOrCode, boolean isColorId) {

```

Comment:
This seems a bit over-complicated to me as a signature, honestly. It seems likely that it's worth the hit to the external method to just move the "getResources()" call to the caller and unify what the method does.